### ISSUE TITLE
(aws-events-targets): Unable to add Dead Letter Queue to BatchJob Event Target

### ISSUE THREAD

**Issue by alanalabouchardiere** [aws/aws-cdk#15238]
CDK does not support the ability to add a dead letter queue to event targets that are batch jobs. This is available in the console, as well as CFN but has yet to be implemented in CDK.

### Use Case

The dead letter queue is useful to determine any issues where batch fails to run. There is not other way to troubleshoot any issues or be alerted if batch fails

### Other

Tried to implement workaround by adding cfn inside of our cdk stack, but while the event rul is created successfully, attaching targets does not work.

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

**Comment by madeline-k** [comment]
Hey @alanalabouchardiere, thanks for opening this issue! I am triaging this as a p2 feature request, which means that the CDK team can't prioritize this in the near future. In the meantime, we are always open to contributions! Feel free to take a look at our [contributing guide](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md), and open a PR. 

You can also use the Cfn-level constructs to access any properties that are available in CloudFormation. I see you said that you've tried that and have encountered an issue. If you share a minimal code sample demonstrating that issue, I might be able to help you find a workaround for this feature request!

**Comment by madeline-k** [comment]
Hey @alanalabouchardiere, thanks for opening this issue! I am triaging this as a p2, which means that the CDK team will not be able to prioritize it in the near future. If you have time to implement this yourself and open a PR, contributions are always welcome! Please checkout the [contributing guide](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md).

You should be able to add a dead letter queue to a Batch Job event target using the Cfn-level classes. If you share more about the issue you are experiencing with that, I may be able to help you with a workaround.

**Comment by alanalabouchardiere** [comment]
Hi @madeline-k thanks for the response! Yes I tried using the Cfn classes and found that while an event rule was created, no target was created in the resulting cfn output from synthing. I can re-write that code if you want an example, but have already removed it.

We ended up going with a lambda between eventbridge and batch as we also needed to provide the full detail json to our batch .net console app. The lambda step has the required dead letter queue, and just does a json->text conversion so that our batch job can utilise it. Hope this makes sense!
